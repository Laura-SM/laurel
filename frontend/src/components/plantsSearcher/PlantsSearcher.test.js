import React from 'react';
import PlantsSearcher from './PlantsSearcher';
import {render} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import CardPlant from '../CardPlant/CardPlant';

describe('Given a PlantsSearcher component', () => {
  const mockStore = configureStore([thunk]);
  const store = mockStore({
    plants: [
      {name: 'kentia', _id: '1a'},
      {name: 'potus', _id: '2a'},
    ],
  });

  describe('When is invoked', () => {
    test('Then it is renders correctly', () => {
      const tree = render(
        <Provider store={store}>
          <PlantsSearcher>
            <CardPlant />
          </PlantsSearcher>
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
