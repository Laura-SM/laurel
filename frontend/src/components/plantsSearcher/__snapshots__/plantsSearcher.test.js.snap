// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a PlantsSearcher component When is invoked Then it is renders correctly 1`] = `
<RCTScrollView
  ListFooterComponent={
    <React.Fragment>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <ForwardRef
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#B8E0D2",
              "borderRadius": 50,
              "height": 60,
              "justifyContent": "center",
              "margin": 20,
              "width": 60,
            }
          }
        >
          <Text>
            Up
          </Text>
        </ForwardRef>
      </View>
    </React.Fragment>
  }
  ListHeaderComponent={
    <React.Fragment>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "margin": 20,
            }
          }
        >
          Find a plant
        </Text>
        <ForwardRef
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#B8E0D2",
              "borderRadius": 50,
              "height": 60,
              "justifyContent": "center",
              "margin": 20,
              "width": 60,
            }
          }
        >
          <Text>
            Fi
          </Text>
        </ForwardRef>
      </View>
    </React.Fragment>
  }
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
              "margin": 20,
            }
          }
        >
          Find a plant
        </Text>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#B8E0D2",
              "borderRadius": 50,
              "height": 60,
              "justifyContent": "center",
              "margin": 20,
              "opacity": 1,
              "width": 60,
            }
          }
        >
          <Text>
            Fi
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#B8E0D2",
              "borderRadius": 50,
              "height": 60,
              "justifyContent": "center",
              "margin": 20,
              "opacity": 1,
              "width": 60,
            }
          }
        >
          <Text>
            Up
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
